src/PowerShellEditorServices/Extensions/EditorObject.cs

```diff
@@ -3,27 +3,60 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 //
 
+using Microsoft.PowerShell.EditorServices.Extensions.Services;
+using Microsoft.PowerShell.EditorServices.Services;
 using System;
 using System.Reflection;
+using System.Threading.Tasks;
 
-namespace Microsoft.PowerShell.EditorServices.Services.PowerShellContext
+namespace Microsoft.PowerShell.EditorServices.Extensions
 {
+    /// <summary>
+    /// Extension class to access the editor API with.
+    /// This is done so that the async/ALC APIs aren't exposed to PowerShell, where they're likely only to cause problems.
+    /// </summary>
+    public static class EditorObjectExtensions
+    {
+        /// <summary>
+        /// Get the provider of extension services for .NET extension tooling.
+        /// </summary>
+        /// <param name="editorObject">The editor object ($psEditor).</param>
+        /// <returns>The extension services provider.</returns>
+        public static EditorExtensionServiceProvider GetExtensionServiceProvider(this EditorObject editorObject)
```

Comment:
This lets you get the API from the editor object, but in a way that the method won't appear to PowerShell (for which it isn't designed)